x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/ClearSecurityCacheRequest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.security.action;
+
+import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.transport.TransportRequest;
+
+import java.io.IOException;
+
+public class ClearSecurityCacheRequest extends BaseNodesRequest<ClearSecurityCacheRequest> {
+
+    private String cacheName;
+    private String[] keys;
+
+    public ClearSecurityCacheRequest() {
+        super((String[]) null);
+    }
+
+    public ClearSecurityCacheRequest(StreamInput in) throws IOException {
+        super(in);
+        cacheName = in.readString();
+        keys = in.readOptionalStringArray();
+    }
+
+    public ClearSecurityCacheRequest cacheName(String cacheName) {
+        this.cacheName = cacheName;
+        return this;
+    }
+
+    public String cacheName() {
+        return cacheName;
+    }
+
+    public ClearSecurityCacheRequest keys(String... keys) {
+        this.keys = keys;
+        return this;
+    }
+
+    public String[] keys() {
+        return keys;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(cacheName);
+        out.writeOptionalStringArray(keys);
+    }
```

Comment:
Ideally we would have serialization tests for every class that has a `writeTo` method.

Reply:
Add a serialisation [test](https://github.com/elastic/elasticsearch/pull/59376/commits/ef95a1dd11410530bccba9d0ddbbb46d50ea4e82#diff-675acc3ee8baa4e58b409727e9b8958c).